### PR TITLE
Source S3: pin CDK and moto

### PR DESCRIPTION
**Pull request by clnoll**
Attempt to fix CATs that have been failing on master since 28 Jan.

Issue for unpinning moto created [here](https://github.com/airbytehq/airbyte-internal-issues/issues/2502).

### CODE REVIEW

File: airbyte-integrations/connectors/source-s3/setup.py [hunk 1]
```diff
@@ -6,15 +6,15 @@
...
+    "airbyte-cdk[file-based]==0.59.2",  # temporarily pin until concurrency can be released
     "smart-open[s3]==5.1.0",
     "wcmatch==8.4",
     "dill==0.3.4",
     "pytz",
     "python-snappy==0.6.1",
 ]
 
-TEST_REQUIREMENTS = ["requests-mock~=1.9.3", "pytest-mock~=3.6.1", "pytest~=6.1", "pandas==2.0.3", "docker", "moto"]
+TEST_REQUIREMENTS = ["requests-mock~=1.9.3", "pytest-mock~=3.6.1", "pytest~=6.1", "pandas==2.0.3", "docker", "moto==4.2.14"]
```
Comment (maxi297) [inline airbyte-integrations/connectors/source-s3/setup.py:17]: Why pin moto?

**Review by maxi297 — COMMENTED** [review]
Small question

File: airbyte-integrations/connectors/source-s3/setup.py (hunk 1 above)
Comment (clnoll) [reply airbyte-integrations/connectors/source-s3/setup.py:17]: We're using something that's been deprecated - `mock_sts`.

Comment (maxi297) [reply airbyte-integrations/connectors/source-s3/setup.py:17]: Do we have a migration plan for this? Does it affect production code paths or only tests?

**Review by maxi297 — COMMENTED** [review]
It seems like CATs are still failing with `botocore.exceptions.NoCredentialsError: Unable to locate credentials`

File: airbyte-integrations/connectors/source-s3/setup.py (hunk 1 above)
Comment (clnoll) [reply airbyte-integrations/connectors/source-s3/setup.py:17]: It just affects tests. There is no migration plan yet but if I can actually get a green build I will make a ticket.

**Review by maxi297 — APPROVED** [review]
Can we add a comment to explain why we pin moto to make sure the knowledge don't get lost. Apart from that, I'm all in for this PR

**Pull request merged**